<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>RegexToolClass</class>
 <widget class="QWidget" name="RegexToolClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>800</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1200</width>
    <height>800</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>RegexTool</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_4">
   <item row="0" column="0">
    <layout class="QGridLayout" name="gridLayout" columnstretch="1,1">
     <property name="leftMargin">
      <number>10</number>
     </property>
     <property name="topMargin">
      <number>10</number>
     </property>
     <property name="rightMargin">
      <number>10</number>
     </property>
     <property name="bottomMargin">
      <number>10</number>
     </property>
     <property name="spacing">
      <number>10</number>
     </property>
     <item row="2" column="1">
      <widget class="QGroupBox" name="groupBox3">
       <property name="title">
        <string>常用正则表达式</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_2">
        <property name="leftMargin">
         <number>10</number>
        </property>
        <property name="topMargin">
         <number>10</number>
        </property>
        <property name="rightMargin">
         <number>10</number>
        </property>
        <property name="bottomMargin">
         <number>10</number>
        </property>
        <property name="spacing">
         <number>10</number>
        </property>
        <item row="0" column="0">
         <layout class="QGridLayout" name="gridLayout2" rowstretch="0,0,0,0,0" columnstretch="1,100">
          <property name="leftMargin">
           <number>10</number>
          </property>
          <property name="topMargin">
           <number>10</number>
          </property>
          <property name="rightMargin">
           <number>10</number>
          </property>
          <property name="bottomMargin">
           <number>10</number>
          </property>
          <property name="horizontalSpacing">
           <number>10</number>
          </property>
          <property name="verticalSpacing">
           <number>20</number>
          </property>
          <item row="3" column="1">
           <widget class="QLabel" name="label4_zh">
            <property name="text">
             <string>[u4e00-u9fa5] or [一-龥] or [\x{4e00}-\x{9fa5}]</string>
            </property>
            <property name="textInteractionFlags">
             <set>Qt::TextSelectableByMouse</set>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>手机号码</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QLabel" name="label3_id">
            <property name="text">
             <string>(^d{15}$)|(^d{18}$)|(^d{17}(d|X|x)$)</string>
            </property>
            <property name="wordWrap">
             <bool>false</bool>
            </property>
            <property name="textInteractionFlags">
             <set>Qt::TextSelectableByMouse</set>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLabel" name="label1_email">
            <property name="text">
             <string>^w+([-+.]w+)*@w+([-.]w+)*.w+([-.]w+)*$</string>
            </property>
            <property name="wordWrap">
             <bool>false</bool>
            </property>
            <property name="textInteractionFlags">
             <set>Qt::TextSelectableByMouse</set>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>中文字符</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>身份证号</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Email地址</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLabel" name="label2_tel">
            <property name="text">
             <string>^(13[0-9]|14[5|7]|15[0|1|2|3|4|5|6|7|8|9]|18[0|1|2|3|5|6|7|8|9])d{8}$</string>
            </property>
            <property name="wordWrap">
             <bool>false</bool>
            </property>
            <property name="textInteractionFlags">
             <set>Qt::TextSelectableByMouse</set>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="label_11">
            <property name="text">
             <string>非负整数</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QLabel" name="label_12">
            <property name="text">
             <string>^[1-9]d*|0$</string>
            </property>
            <property name="textInteractionFlags">
             <set>Qt::LinksAccessibleByMouse|Qt::TextSelectableByMouse</set>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QGroupBox" name="groupBox1">
       <property name="title">
        <string>相关文档</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_3">
        <property name="leftMargin">
         <number>10</number>
        </property>
        <property name="topMargin">
         <number>10</number>
        </property>
        <property name="rightMargin">
         <number>10</number>
        </property>
        <property name="bottomMargin">
         <number>10</number>
        </property>
        <property name="spacing">
         <number>10</number>
        </property>
        <item row="0" column="0">
         <layout class="QGridLayout" name="gridLayout1" rowstretch="0,0,0" columnstretch="1,100">
          <property name="leftMargin">
           <number>10</number>
          </property>
          <property name="topMargin">
           <number>10</number>
          </property>
          <property name="rightMargin">
           <number>10</number>
          </property>
          <property name="bottomMargin">
           <number>10</number>
          </property>
          <property name="horizontalSpacing">
           <number>10</number>
          </property>
          <property name="verticalSpacing">
           <number>20</number>
          </property>
          <item row="1" column="0">
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string>OSCHINA在线正则表达式测试</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLabel" name="label1_email_2">
            <property name="text">
             <string>&lt;a href=&quot;https://perldoc.perl.org/perlre&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#0000ff;&quot;&gt;https://perldoc.perl.org/perlre&lt;/span&gt;&lt;/a&gt;</string>
            </property>
            <property name="openExternalLinks">
             <bool>true</bool>
            </property>
            <property name="textInteractionFlags">
             <set>Qt::LinksAccessibleByMouse|Qt::TextSelectableByMouse</set>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>Perl regular expressions</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLabel" name="label_8">
            <property name="text">
             <string>&lt;a href=&quot;https://tool.oschina.net/regex&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#0000ff;&quot;&gt;https://tool.oschina.net/regex&lt;/span&gt;&lt;/a&gt;</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_9">
            <property name="text">
             <string>菜鸟教程 - 正则表达式</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QLabel" name="label_10">
            <property name="text">
             <string>&lt;a href=&quot;https://www.runoob.com/regexp/regexp-tutorial.html&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#0000ff;&quot;&gt;https://www.runoob.com/regexp/regexp-tutorial.html&lt;/span&gt;&lt;/a&gt;</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item row="0" column="0" rowspan="4">
      <layout class="QGridLayout" name="gridLayout0">
       <item row="1" column="2">
        <widget class="QPushButton" name="btn_match">
         <property name="text">
          <string>匹配</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0" colspan="3">
        <widget class="QTextEdit" name="te_input">
         <property name="placeholderText">
          <string>在此输入待匹配文本</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0" colspan="3">
        <widget class="QTextEdit" name="te_output">
         <property name="readOnly">
          <bool>true</bool>
         </property>
         <property name="placeholderText">
          <string>匹配结果……</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLineEdit" name="line_regex">
         <property name="placeholderText">
          <string>在此输入正则表达式</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label">
         <property name="text">
          <string>正则表达式</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="1" column="1">
      <widget class="QGroupBox" name="groupBox2">
       <property name="title">
        <string>语法参考</string>
       </property>
       <layout class="QFormLayout" name="formLayout">
        <property name="horizontalSpacing">
         <number>10</number>
        </property>
        <property name="verticalSpacing">
         <number>10</number>
        </property>
        <property name="leftMargin">
         <number>10</number>
        </property>
        <property name="topMargin">
         <number>10</number>
        </property>
        <property name="rightMargin">
         <number>10</number>
        </property>
        <property name="bottomMargin">
         <number>10</number>
        </property>
        <item row="0" column="1">
         <widget class="QTableWidget" name="tableWidget">
          <property name="editTriggers">
           <set>QAbstractItemView::NoEditTriggers</set>
          </property>
          <property name="selectionMode">
           <enum>QAbstractItemView::NoSelection</enum>
          </property>
          <attribute name="horizontalHeaderVisible">
           <bool>false</bool>
          </attribute>
          <attribute name="horizontalHeaderMinimumSectionSize">
           <number>120</number>
          </attribute>
          <attribute name="horizontalHeaderDefaultSectionSize">
           <number>120</number>
          </attribute>
          <attribute name="horizontalHeaderStretchLastSection">
           <bool>true</bool>
          </attribute>
          <attribute name="verticalHeaderVisible">
           <bool>false</bool>
          </attribute>
          <attribute name="verticalHeaderMinimumSectionSize">
           <number>36</number>
          </attribute>
          <attribute name="verticalHeaderDefaultSectionSize">
           <number>36</number>
          </attribute>
          <row>
           <property name="text">
            <string>0</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>1</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>2</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>3</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>4</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>5</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>6</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>7</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>8</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>9</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>10</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>11</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>12</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>13</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>14</string>
           </property>
          </row>
          <column>
           <property name="text">
            <string>语法</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>含义</string>
           </property>
          </column>
          <item row="0" column="0">
           <property name="text">
            <string>.</string>
           </property>
          </item>
          <item row="0" column="1">
           <property name="text">
            <string>除换行符以外的所有字符</string>
           </property>
          </item>
          <item row="1" column="0">
           <property name="text">
            <string>^</string>
           </property>
          </item>
          <item row="1" column="1">
           <property name="text">
            <string>字符串开头</string>
           </property>
          </item>
          <item row="2" column="0">
           <property name="text">
            <string>$</string>
           </property>
          </item>
          <item row="2" column="1">
           <property name="text">
            <string>字符串结尾</string>
           </property>
          </item>
          <item row="3" column="0">
           <property name="text">
            <string>\d, \w, \s</string>
           </property>
          </item>
          <item row="3" column="1">
           <property name="text">
            <string>匹配数字、字符、空格</string>
           </property>
          </item>
          <item row="4" column="0">
           <property name="text">
            <string>\D, \W, \S</string>
           </property>
          </item>
          <item row="4" column="1">
           <property name="text">
            <string>匹配非数字、非字符、非空格</string>
           </property>
          </item>
          <item row="5" column="0">
           <property name="text">
            <string>[abc]</string>
           </property>
          </item>
          <item row="5" column="1">
           <property name="text">
            <string>匹配 a、b 或 c 中的一个字母</string>
           </property>
          </item>
          <item row="6" column="0">
           <property name="text">
            <string>[a-z]</string>
           </property>
          </item>
          <item row="6" column="1">
           <property name="text">
            <string>匹配 a 到 z 中的一个字母</string>
           </property>
          </item>
          <item row="7" column="0">
           <property name="text">
            <string>aa|bb</string>
           </property>
          </item>
          <item row="7" column="1">
           <property name="text">
            <string>匹配 aa 或 bb</string>
           </property>
          </item>
          <item row="8" column="0">
           <property name="text">
            <string>?</string>
           </property>
          </item>
          <item row="8" column="1">
           <property name="text">
            <string>匹配0次或1次</string>
           </property>
          </item>
          <item row="9" column="0">
           <property name="text">
            <string>+</string>
           </property>
          </item>
          <item row="9" column="1">
           <property name="text">
            <string>匹配1次或多次</string>
           </property>
          </item>
          <item row="10" column="0">
           <property name="text">
            <string>*</string>
           </property>
          </item>
          <item row="10" column="1">
           <property name="text">
            <string>匹配0次或多次</string>
           </property>
          </item>
          <item row="11" column="0">
           <property name="text">
            <string>{n}</string>
           </property>
          </item>
          <item row="11" column="1">
           <property name="text">
            <string>匹配n次</string>
           </property>
          </item>
          <item row="12" column="0">
           <property name="text">
            <string>{n,}</string>
           </property>
          </item>
          <item row="12" column="1">
           <property name="text">
            <string>匹配至少n次</string>
           </property>
          </item>
          <item row="13" column="0">
           <property name="text">
            <string>{,n}</string>
           </property>
          </item>
          <item row="13" column="1">
           <property name="text">
            <string>匹配最多n次</string>
           </property>
          </item>
          <item row="14" column="0">
           <property name="text">
            <string>{n,m}</string>
           </property>
          </item>
          <item row="14" column="1">
           <property name="text">
            <string>匹配至少n次但不超过m次</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="regex_tool.qrc"/>
 </resources>
 <connections/>
</ui>
